src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/services/NodeService.kt

```diff
@@ -27,9 +29,11 @@ class NodeService(
   @Autowired private val transactionsEventStoreRepository: TransactionsEventStoreRepository<Any>
 ) {
   var logger: Logger = LoggerFactory.getLogger(NodeService::class.java)
+
   suspend fun closePayment(
     transactionId: UUID,
-    transactionOutcome: ClosePaymentRequestV2Dto.OutcomeEnum
+    transactionOutcome: ClosePaymentRequestV2Dto.OutcomeEnum,
+    authorizationCode: Optional<String>
```

Comment:
There is one thing I don't like too much about the method's signature: there is nothing telling me that some parameter combinations are not allowed:
 * outcome OK, some auth code  ✅ 
 * outocome OK, no auth code  ❌ 
 * outcome KO, some auth code  ❌ 
 * outcome KO, no auth code ✅

What about accepting only an auth code and deriving the outcome from it? Or, alternatively, having a tiny class to just hold these parameters and allow construction only of these cases, e.g.

```kotlin
/* sketch */
class ClosePaymentParams {
    public val outcome: ClosePaymentRequestV2Dto.OutcomeEnum
    public val authorizationCode: Optional<String>

    constructor(authorizationCode: Optional<String>) {
        this.authorizationCode = authorizationCode
        this.outcome = if (authorizationCode.isEmpty()) { ClosePaymentRequestV2Dto.OutcomeEnum.KO } else { ClosePaymentRequestV2Dto.OutcomeEnum.OK }
    }
}
```

and use that in the method

Reply:
I don't like this solution. It is true that there's a dependency between the outcome and the authorizationCode, but in those both ways, we mask the authorization response and set the OK or KO depending of the authorizationCode presence or not, and not mapping the value from the auth response.

If we had a dirty response like {outcome: AuhtorizationResultCode.OK, authorizationCode: null} we will set {KO,empty} for this parameter, and it could be really very hard to get the dirty result from nodo.